src/onnx/parse_dynamicquantizelinear.cpp

```diff
@@ -98,41 +98,40 @@ struct parse_dynamicquantizelinear : op_parser<parse_dynamicquantizelinear>
 
-        auto x_reshaped =
-            (x_shape.lens().size() == 1)
-                ? x
-                : info.add_instruction(
-                      migraphx::make_op("reshape", {{"dims", {x_shape.elements()}}}), x);
-
         auto lit_0 = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {0}});
-        x_reshaped =
-            info.add_instruction(migraphx::make_op("concat", {{"axis", 0}}), x_reshaped, lit_0);
-
         // 1. Computing y_scale
         // Note: currently, DynamicQuantizeLinear only has uint8 quantization:
-        const auto x_max = std::numeric_limits<uint8_t>::max();
-        const auto x_min = std::numeric_limits<uint8_t>::min();
-
-        auto q_range =
-            info.add_literal(migraphx::literal{migraphx::shape{x_type}, {x_max - x_min}});
+        const auto type_max = std::numeric_limits<uint8_t>::max();
+        const auto type_min = std::numeric_limits<uint8_t>::min();
+        std::vector<size_t> len_vec(x_shape.lens().size());
+        std::iota(len_vec.begin(), len_vec.end(), 0);
 
         // maximum(0, max(x))
-        auto max_x =
-            info.add_instruction(migraphx::make_op("reduce_max", {{"axes", {0}}}), x_reshaped);
+        auto reduce_max_x =
+            info.add_instruction(migraphx::make_op("reduce_max", {{"axes", len_vec}}), x);
+        auto max_x = info.add_common_op("max", lit_0, reduce_max_x);
+
         // minimum(0, min(x))
-        auto min_x =
-            info.add_instruction(migraphx::make_op("reduce_min", {{"axes", {0}}}), x_reshaped);
+        auto reduce_min_x =
+            info.add_instruction(migraphx::make_op("reduce_min", {{"axes", len_vec}}), x);
+        auto min_x = info.add_common_op("min", lit_0, reduce_min_x);
+
+        auto q_range = info.add_literal(migraphx::literal{
+            migraphx::shape{x_type, max_x->get_shape().lens()}, {type_max - type_min}});
+
+        auto q_min = info.add_literal(
+            migraphx::literal{migraphx::shape{x_type, min_x->get_shape().lens()}, {type_min}});
+        auto q_max = info.add_literal(
+            migraphx::literal{migraphx::shape{x_type, max_x->get_shape().lens()}, {type_max}});
 
         // y_scale = (maximum(0, max(x)) - minimum(0, min(x))) / (qmax - qmin)
         auto sub0    = info.add_common_op("sub", max_x, min_x);
         auto y_scale = info.add_common_op("div", sub0, q_range);
 
         // 2. Computing y_zero_point
         // intermediate_zero_point = qmin - min(x) / y_scale
-        auto q_min     = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {x_min}});
-        auto q_max     = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {x_max}});
-        auto sub1      = info.add_common_op("sub", q_min, min_x);
-        auto interm_zp = info.add_common_op("div", sub1, y_scale);
+        auto div1      = info.add_common_op("div", min_x, y_scale);
+        auto interm_zp = info.add_common_op("sub", q_min, div1);
```

Comment:
Shouldn't the Zero Point be also readjusted in simplify_dynamicquantizelinear.cpp?

Reply:
I think this is fine..